ai2thor/controller.py

```diff
@@ -410,6 +411,22 @@ def __init__(
         self.add_depth_noise = add_depth_noise
         self.include_private_scenes = include_private_scenes
         self.x_display = None
+        self.gpu_device = gpu_device
+        cuda_visible_devices = list(
+            map(int, filter(lambda y: y.isdigit(),
+            map(lambda x: x.strip(),
+                os.environ.get('CUDA_VISIBLE_DEVICES', '').split(',')))))
+
+        if self.gpu_device:
+            if (type(self.gpu_device) is not int or self.gpu_device < 0):
```

Comment:
`not np.issubdtype(type(self.gpu_device), np.integer)` might be safer if someone passes in a numpy int.

Reply:
Fixed.  Changed type check to use the `numbers` module.